clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -359,39 +359,38 @@ public ConsumerRebalanceListener listener() {
     }
 
     private static class TopicPartitionState {
-        private Long position;
+        private Long position; // last consumed position
         private OffsetAndMetadata committed;  // last committed position
-
-        private boolean hasValidPosition; // whether we have valid consumed and fetched positions
         private boolean paused;  // whether this partition has been paused by the user
-        private boolean awaitingReset; // whether we are awaiting reset
-        private OffsetResetStrategy resetStrategy;  // the reset strategy if awaitingReset is set
+        private OffsetResetStrategy resetStrategy;  // the strategy to use if the offset needs resetting
 
         public TopicPartitionState() {
             this.paused = false;
             this.position = null;
             this.committed = null;
-            this.awaitingReset = false;
-            this.hasValidPosition = false;
             this.resetStrategy = null;
         }
 
         private void awaitReset(OffsetResetStrategy strategy) {
-            this.awaitingReset = true;
             this.resetStrategy = strategy;
             this.position = null;
-            this.hasValidPosition = false;
+        }
+
+        public boolean awaitingReset() {
+            return resetStrategy != null;
+        }
+
+        public boolean hasValidPosition() {
+            return position != null;
         }
 
         private void seek(long offset) {
             this.position = offset;
-            this.awaitingReset = false;
             this.resetStrategy = null;
-            this.hasValidPosition = true;
         }
 
         private void position(long offset) {
-            if (!hasValidPosition)
+            if (position == null)
```

Comment:
Any hurt to reuse `hasValidPosition()` here? IMO, it makes the semantics more clear.